clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1284,14 +1372,23 @@ class SyclKernelDeclCreator : public SyclKernelFieldHandler {
     // Don't do -1 here because we count on this to be the first parameter added
     // (if any).
     size_t ParamIndex = Params.size();
-    for (const ParmVarDecl *Param : InitMethod->parameters())
-      addParam(BS, Param->getType().getCanonicalType());
+    auto ParamIt = InitMethod->parameters().begin();
+    if (*ParamIt) {
+      // Add meaningful argument (not '-1') to buffer_location attribute only
+      // for an accessor pointer
+      size_t BufferLocAttrArg =
+          handleBufferLocationProperty(FieldTy, BS.getBeginLoc());
```

Comment:
BTW, I haven't seen any test cases with inheritance (this function handles inherited accessors).

Reply:
Added the tests, actually I wasn't sure why do we have `handleSyclAccessorType` function if we already have `handleSyclSpecialType`, so thanks for pointing on it out!